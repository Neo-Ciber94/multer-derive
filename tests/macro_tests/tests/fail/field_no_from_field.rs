use multer_derive::FromMultipart;

#[derive(FromMultipart)]
struct MyStruct {
    text: String,
    other: OtherType
}

struct OtherType {
    value: String
}

fn main() {}
