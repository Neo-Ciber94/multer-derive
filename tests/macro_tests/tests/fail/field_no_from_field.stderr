error[E0277]: the trait bound `OtherType: FromMultipartField` is not satisfied
 --> tests/fail/field_no_from_field.rs:3:10
  |
3 | #[derive(FromMultipart)]
  |          ^^^^^^^^^^^^^ the trait `FromMultipartField` is not implemented for `OtherType`
  |
  = help: the following other types implement trait `FromMultipartField`:
            &'a PhantomData<T>
            ()
            (K, V)
            Arc<T>
            AtomicBool
            AtomicI16
            AtomicI32
            AtomicI64
          and $N others
  = note: required for `OtherType` to implement `FromMultipart`
  = note: this error originates in the derive macro `FromMultipart` (in Nightly builds, run with -Z macro-backtrace for more info)
